### PR TITLE
feat: automatic tileset updates

### PR DESCRIPTION
**Pull request by casswedson**
#### Summary
None

#### Purpose of change
Part of #57559, takes the scrip presented there and uses it to make a pr with the changes it produces

There are more than a few problems with running a scrip on demand for `tiles` releases on multiple platforms, mainly tools are not available on environment, tool is from a different version thus it does not work for the script's purpose. uhh too hard

Similar to the translation updates, this automates tileset updates, at least from the ones <https://github.com/I-am-Erk/CDDA-Tilesets> this repo has, mind you not all tilesets are released giving us some control over what tileset is obsolete

Also update Cuteclysm since it implemented automatic releases <https://github.com/pixel-32/CDDA-tileset>

- [x] runs at 23:00 UTC sundays, few hours after the actual release upstream, gives us time to make fixes if they are needed
- [x] only make prs against the cdda repo

~~This isn't good to go at the moment cause there are a few errors upstream~~ all good upstream, all good here

#### Describe the solution
```yml
          # download the newest released zipped composed tileset from these repos
          gh release download -D gfx/ -p '*.zip' -R I-am-Erk/CDDA-Tilesets
          gh release download -D gfx/ -p '*.zip' -R pixel-32/CDDA-Tileset
          # unzip them and replace the ones in the gfx folder
          find gfx/ -name "*.zip" | xargs -I{} unzip -o {} -d gfx/
          # zip files are ignored but remove them anyway
          rm -fr gfx/*.zip
          # make sure the downloaded json is linted with cdda's json formmater
          # by default the composer just spits out a loooong line of json it's usable buth ehh
          make tools/format/json_formatter.cgi
          find gfx/ -name "*.json" -print0 | xargs -0 -L 1 -P $(nproc) tools/format/json_formatter.cgi || exit 0
```

#### Testing
<https://github.com/casswedson/Cataclysm-DDA/pull/29>

### CODE REVIEW

File: .github/workflows/update-tilesets.yml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+          curl -s https://api.github.com/repos/I-am-Erk/CDDA-Tilesets/releases/latest | grep "browser_download_url.*zip" | cut -d : -f 2,3 | tr -d \" | wget -P gfx -i -
+          find gfx/ -name "*.zip" | xargs -I{} unzip -o {} -d gfx/
+          rm -fr gfx/*.zip
+      - name: Create Pull Request
+        uses: peter-evans/create-pull-request@v4
+        with:
+          commit-message: |
+            Routine Tileset updates on ${{ steps.get-timestamp.outputs.time }}
+
+            Co-authored-by: Fris0uman <41293484+Fris0uman@users.noreply.github.com>
```
Comment (Fris0uman) [inline .github/workflows/update-tilesets.yml:30]:
```suggestion
```
It doesn't make sense to list me as a co-author, we can add link to the tileset repos in `tileset.txt` of tilesets that don't have it yet if we want to double down on proper attribution, but the update itself can be authored just by a github action.

**Comment by Fris0uman** [comment]
We should probably use the `--format-json` argument to the tileset build script to reduce the difference between json when updating tilesets. I tried to do it there https://github.com/Fris0uman/CDDA-Tilesets/pull/11 but I get an error when `compose.py` tries to use `json_formatter.cgi`

**Comment by Fris0uman** [comment]
The automatic release of cuteclysm is missing `--format-json` argument I think

**Comment by casswedson** [comment]
That's true, but I also made this:
```sh
          # make sure all files are linted just in case
          make tools/format/json_formatter.cgi
          find gfx/ -name "*.json" -print0 | xargs -0 -L 1 -P $(nproc) tools/format/json_formatter.cgi || exit 0
```
this makes sure whatever comes from the upstream releases is linted by the time the pr is made [here check the log for the action](https://github.com/casswedson/Cataclysm-DDA/runs/6381377486?check_suite_focus=true) the other thing I could do is implement the cdda json formatter in every upstream repo and that would be more than 2 lines, so this is better, simpler

**Pull request merged**